client/test/service.js

```diff
@@ -32,6 +35,79 @@ const importUpload = async (request) => {
   }
 }
 
+/**
+ * @param {File} file
+ * @returns {Promise<CID>}
+ */
+const importAsset = async (file) => {
+  const { cid } = await importDirectory([file])
+  return CID.parse(cid.toString())
+}
+
+/**
+ * @param {Request} request
+ */
+const importToken = async (request) => {
+  const contentType = request.headers.get('content-type') || ''
+  if (contentType.includes('multipart/form-data')) {
+    const form = await request.formData()
+
+    const data = JSON.parse(/** @type {string} */ (form.get('meta')))
+    const dag = JSON.parse(JSON.stringify(data))
+    const metadata = JSON.parse(JSON.stringify(data))
+
+    for (const [path, content] of form.entries()) {
+      if (path !== 'meta') {
+        const file = /** @type {File} */ (content)
+        const cid = await importAsset(file)
+        const href = `ipfs://${cid}/${file.name}`
+        setAt(path.split('.'), dag, cid)
+        setAt(path.split('.'), data, { '@': 'URL', href })
+        setAt(path.split('.'), metadata, href)
+      }
+    }
+
+    dag.meta = await importAsset(
+      new File([JSON.stringify(metadata)], 'data.json')
+    )
+
+    const bytes = CBOR.encode(dag)
+    const hash = await sha256.digest(bytes)
+    const cid = CID.create(1, CBOR.code, hash)
+
+    const result = {
+      ok: true,
+      value: {
+        ipld: cid.toString(),
+        metadata: { '@': 'URL', href: `ipfs://${dag.meta}/data.json` },
+        data,
+      },
+    }
+
+    return result
+  } else {
+    throw Error('/api/store expects multipart/form-data')
+  }
+}
+
+/**
+ * @template V
```

Comment:
```suggestion
 * Sets a given `value` at the given `path` on a passed `object`.
 *
 * @example
 * ```js
 * const obj = { a: { b: { c: 1 }}}
 * setAt('a.b.c', obj, 5)
 * obj.a.b.c //> 5
 * ```
 * 
 * @template V
```